expression/builtin_miscellaneous.go

```diff
@@ -176,23 +183,130 @@ func (c *anyValueFunctionClass) getFunction(args []Expression, ctx context.Conte
 	if err := c.verifyArgs(args); err != nil {
 		return nil, errors.Trace(err)
 	}
-	sig := &builtinAnyValueSig{newBaseBuiltinFunc(args, ctx)}
+	argTp := fieldTp2EvalTp(args[0].GetType())
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, argTp, argTp)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	*bf.tp = *args[0].GetType()
+	var sig builtinFunc
+	switch argTp {
+	case tpDecimal:
+		sig = &builtinDecimalAnyValueSig{baseDecimalBuiltinFunc{bf}}
+	case tpDuration:
+		sig = &builtinDurationAnyValueSig{baseDurationBuiltinFunc{bf}}
+	case tpInt:
+		sig = &builtinIntAnyValueSig{baseIntBuiltinFunc{bf}}
+	case tpJSON:
+		sig = &builtinJSONAnyValueSig{baseJSONBuiltinFunc{bf}}
+	case tpReal:
+		sig = &builtinRealAnyValueSig{baseRealBuiltinFunc{bf}}
+	case tpString:
+		sig = &builtinStringAnyValueSig{baseStringBuiltinFunc{bf}}
+	case tpTime:
+		sig = &builtinTimeAnyValueSig{baseTimeBuiltinFunc{bf}}
+	default:
+		panic("unexpected evalTp")
```

Comment:
panic message can be more specific, like: "unexpected evalTp of builtin function ANY_VALUE"